File: python/endpoints/file_reader.py
from typing import Dict, Any, List
from fastapi import APIRouter
from file_readers import read_yaml_file, read_txt_file, read_xml_file, read_json_file, read_csv_file

router = APIRouter()

@router.get("/csvFile", response_model=Dict)
async def get_csv_file() -> Dict:
    csv_content = read_csv_file('../sample.csv')
    return csv_content

@router.get("/jsonFile", response_model=Any)
async def get_json_file() -> Any:
    json_content = read_json_file('../sample.json')
    return json_content

@router.get("/txtFile", response_model=str)
async def get_txt_file() -> str:
    txt_content = read_txt_file('../sample.txt')
    return txt_content

@router.get("/xmlFile", response_model=List)
async def get_xml_file() -> List[str]:
    xml_content = read_xml_file('../sample.xml', 'child')
    return xml_content

@router.get("/yamlFile", response_model=Any)
async def get_yaml_file() -> Any:
    yaml_content = read_yaml_file('../sample.yaml')
    return yaml_content

File: javascript/endpoints/fileReader.js
import express from 'express';
import { readCsvFile } from '../fileReaders/readCsvFile.js';
import { readJsonFile } from '../fileReaders/readJsonFile.js';
import { readTextFile } from '../fileReaders/readTextFile.js';
import { readXmlFile } from '../fileReaders/readXmlFile.js';

const app = express();
const port = 3000;

app.get('/read-csv', async (req, res) => {
    try {
        const path = '../sample.csv';
        const csvData = await readCsvFile(path);
        res.json(csvData);
    } catch (err) {
        console.error('Failed to read CSV:', err);
        res.status(500).send('Failed to read CSV file.');
    }
});

app.get('/read-json', async (req, res) => {
    try {
        const path = '../sample.json';
        const jsonData = await readJsonFile(path);
        res.json(jsonData);
    } catch (err) {
        console.error('Failed to read JSON:', err);
        res.status(500).send('Failed to process JSON file.');
    }
});

app.get('/read-text', async (req, res) => {
    try {
        const path = '../sample.txt';
        const textData = await readTextFile(path);
        res.send(textData);
    } catch (err) {
        console.error('Failed to read text file:', err);
        res.status(500).send('Failed to process text file.');
    }
});

app.get('/read-xml', async (req, res) => {
    try {
      const path = '../sample.xml'; 
      const xmlData = await readXmlFile(path);
      res.json(xmlData);
    } catch (err) {
      console.error('Failed to read XML file:', err);
      res.status(500).send('Failed to process XML file.');
    }
  });

  app.get('/read-yaml', async (req, res) => {
    try {
      const path = '../sample.yaml';
      const yamlData = await readYamlFile(path);
      res.json(yamlData);
    } catch (err) {
      console.error('Failed to read YAML file:', err);
      res.status(500).send('Failed to process YAML file.');
    }
  });

  app.get('/requestFastAPI', async (req, res) => {
    const response = await fetch('http://127.0.0.1:8000/jsonFile', {
        method: 'GET',
    });
    const result = await response.json();
    res.send({ data: result });
});

app.listen(port, () => {
    console.log(`Server running at http://localhost:${port}`);
});
